yt_dlp/extractor/parti.py

```diff
@@ -0,0 +1,114 @@
+                'is_live': False,
+                'categories': ['Gaming'],
+                'thumbnail': 'https://assets.parti.com/351424_eb9e5250-2821-484a-9c5f-ca99aa666c87.png',
+                'channel': 'ItZTMGG',
+                'timestamp': 1743044379,
+            },
+            'params': {'skip_download': 'm3u8'},
+        },
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        data = self._call_api(f'get_livestream_channel_info/recent/{video_id}', video_id)
+        formats = self._extract_m3u8_formats(
+            urljoin('https://watch.parti.com', data['livestream_recording']), video_id, 'mp4')
+
+        return {
+            'id': video_id,
+            'formats': formats,
+            'is_live': False,
+            **traverse_obj(data, {
+                'title': ('event_title', {str}),
+                'channel': ('user_name', {str}),
+                'thumbnail': ('event_file', {url_or_none}),
+                'categories': ('category_name', {str}, all),
+                'timestamp': ('event_start_ts', {int_or_none}),
+            }),
+        }
+
+
+class PartiLivestreamIE(PartiBaseIE):
+    IE_NAME = 'parti:livestream'
+    _VALID_URL = r'https?://(?:www\.)?parti\.com/creator/(?P<service>[\w]+)/(?P<id>[\w/-]+)'
+    _TESTS = [
+        {
+            'url': 'https://parti.com/creator/parti/Capt_Robs_Adventures',
+            'info_dict': {
+                'id': 'Capt_Robs_Adventures',
+                'ext': 'mp4',
+                'title': str,
+                'description': str,
+                'upload_date': str,
+                'is_live': True,
+                'live_status': 'is_live',
+            },
+            'params': {'skip_download': 'm3u8'},
+        },
+        {
+            'url': 'https://parti.com/creator/discord/sazboxgaming/0',
+            'only_matching': True,
+        },
+    ]
+
+    def _real_extract(self, url):
+        service, creator_slug = self._match_valid_url(url).group('service', 'id')
+
+        encoded_creator_slug = creator_slug.replace('/', '%23')
+        creator_id = self._call_api(
+            f'get_user_by_social_media/{service}/{encoded_creator_slug}',
+            creator_slug, note='Fetching user ID')
+
+        data = self._call_api(
+            f'get_livestream_channel_info/{creator_id}', creator_id,
+            note='Fetching user profile feed')['channel_info']
+
+        if not traverse_obj(data, ('channel', 'is_live', {bool})):
+            raise UserNotLive(video_id=creator_id)
+
+        stream_data = traverse_obj(data, ('channel', {
+            'url': ('playback_url', {url_or_none}, {require('Stream url')}),
+            'token': ('playback_auth_token', {str}, {require('Stream token')}),
+        }))
+
+        formats = self._extract_m3u8_formats(
+            stream_data['url'], creator_slug, live=True, query={
+                'token': stream_data['token'],
+                'player_version': '1.17.0',
+            })
```

Comment:
If both the url and token are required, we don't need the traversal/arbitrary dict here, we can just let the KeyErrors rip
```suggestion
        channel_info = data['channel']
        formats = self._extract_m3u8_formats(
            channel_info['playback_url'], creator_slug, live=True, query={
                'token': channel_info['playback_auth_token'],
                'player_version': '1.17.0',
            })
```